d2g.go

```diff
@@ -150,31 +169,41 @@ func command(payload *dockerworker.DockerWorkerPayload, dwImage Image, gwArtifac
 	}, nil
 }
 
-func podmanRunCommand(containerName string, payload *dockerworker.DockerWorkerPayload, dwImage Image, wdcs []genericworker.WritableDirectoryCache) (string, error) {
+func podmanRunCommand(containerName string, dwPayload *dockerworker.DockerWorkerPayload, dwImage Image, wdcs []genericworker.WritableDirectoryCache) (string, error) {
 	command := strings.Builder{}
 	command.WriteString("podman run --name " + containerName)
-	if payload.Capabilities.Privileged {
+	if dwPayload.Capabilities.Privileged {
 		command.WriteString(" --privileged")
 	}
-	if payload.Features.AllowPtrace {
+	if dwPayload.Features.AllowPtrace {
 		command.WriteString(" --cap-add=SYS_PTRACE")
 	}
-	command.WriteString(createVolumeMountsString(payload.Cache, wdcs))
+	command.WriteString(createVolumeMountsString(dwPayload.Cache, wdcs))
 	command.WriteString(" --add-host=taskcluster:127.0.0.1 --net=host")
-	command.WriteString(podmanEnvMappings(payload.Env))
+	command.WriteString(podmanEnvMappings(dwPayload.Env))
 	dockerImageString, err := dwImage.String()
 	if err != nil {
 		return "", fmt.Errorf("could not form docker image string: %w", err)
 	}
 	command.WriteString(" " + dockerImageString)
-	command.WriteString(" " + shell.Escape(payload.Command...))
+	command.WriteString(" " + shell.Escape(dwPayload.Command...))
 	return command.String(), nil
 }
 
-func podmanCopyArtifacts(containerName string, payload *dockerworker.DockerWorkerPayload, gwArtifacts []genericworker.Artifact) []string {
-	commands := make([]string, len(gwArtifacts))
+func podmanCopyArtifacts(containerName string, dwPayload *dockerworker.DockerWorkerPayload, gwArtifacts []genericworker.Artifact) []string {
+	commands := []string{}
 	for i := range gwArtifacts {
-		commands[i] = "podman cp '" + containerName + ":" + payload.Artifacts[gwArtifacts[i].Name].Path + "' " + gwArtifacts[i].Path
+		// An image artifact will be in the generic worker payload
+		// when dockerSave is enabled. That artifact will not be
+		// found in either the docker worker payload or the container
+		// after the podman run command is complete, so no podman cp
+		// command is needed for it.
+		// The image artifact is created after the podman run
+		// command is complete.
+		if _, ok := dwPayload.Artifacts[gwArtifacts[i].Name]; !ok {
+			continue
+		}
+		commands = append(commands, "podman cp '"+containerName+":"+dwPayload.Artifacts[gwArtifacts[i].Name].Path+"' "+gwArtifacts[i].Path)
```

Comment:
```suggestion
		commands = append(commands, fmt.Sprintf("podman cp '%s:%s' %s", containerName, dwPayload.Artifacts[gwArtifacts[i].Name].Path, gwArtifacts[i].Path))
```

might be a bit easier to read